src/vivarium_csu_hypertension_sdc/components/hypertension_intervention.py

```diff
@@ -0,0 +1,159 @@
+
+    def setup(self, builder):
+        self.clock = builder.time.clock()
+
+        self.sim_start = pd.Timestamp(**builder.configuration.time.start)
+
+        self.config = builder.configuration.hypertension_treatment
+
+        columns_created = ['followup_date', 'last_visit_date', 'last_visit_type',
+                           'high_systolic_blood_pressure_measurement',
+                           'high_systolic_blood_pressure_last_measurement_date']
+        builder.population.initializes_simulants(self.on_initialize_simulants,
+                                                 requires_columns=DOSAGE_COLUMNS,
+                                                 creates_columns=columns_created)
+        self.population_view = builder.population.get_view(DOSAGE_COLUMNS + columns_created + ['alive'],
+                                                           query='alive == "alive"')
+
+        self.randomness = {'followup_scheduling': builder.randomness.get_stream('followup_scheduling'),
+                           'background_visit_attendance': builder.randomness.get_stream('background_visit_attendance'),
+                           'sbp_measurement': builder.randomness.get_stream('sbp_measurement'),
+                           'therapeutic_inertia': builder.randomness.get_stream('therapeutic_intertia')
+                           }
+
+        self.ti_probability = utilities.get_therapeutic_inertia_probability(self.config.therapeutic_inertia.mean,
+                                                                            self.config.therapeutic_inertia.sd,
+                                                                            self.randomness['therapeutic_inertia'])
+
+        self.utilization_data = builder.lookup.build_table(
+            builder.data.load('healthcare_entity.outpatient_visits.utilization_rate'))
+        self.healthcare_utilization = builder.value.register_rate_producer('healthcare_utilization_rate',
+                                                                           source=lambda index: self.utilization_data(index))
+
+        self.sbp = builder.value.get_value('high_systolic_blood_pressure.exposure')
+
+        builder.event.register_listener('time_step', self.on_time_step)
+
+    def on_initialize_simulants(self, pop_data):
+        drug_dosages = self.population_view.subview(DOSAGE_COLUMNS).get(pop_data.index)
+        sims_on_tx = drug_dosages.loc[drug_dosages.sum(axis=1) > 0].index
+
+        initialize = pd.DataFrame({'followup_date': pd.NaT, 'last_visit_date': pd.NaT, 'last_visit_type': None,
+                                   'high_systolic_blood_pressure_measurement': np.nan,
+                                   'high_systolic_blood_pressure_last_measurement_date': pd.NaT},
+                                  index=pop_data.index)
+
+        durations = utilities.get_durations_in_range(self.randomness['followup_scheduling'],
+                                           low=0, high=3*28,
```

Comment:
Pull out your follow up duration into a global variable or configuration parameter (if you can do a parameter, that's better).

Reply:
will do